### DISCUSSION TITLE
NextJS Deployment using Terraform

### DISCUSSION THREAD

**Discussion by bamofah** [vercel/next.js#45669] [Help]
Hi,
I am currently exploring adopting Nextjs with terraform. Currently have a terraform based project that is using React.JS static assets to S3. I would like to switch over to Next.js to start benefiting of some of the features like SSR and wanted to know if there is any support fr it or example that can help me on that journey. I know on your site one of the deployment mechanisms mentions this repo [terraform-aws-next](https://github.com/milliHQ/terraform-aws-next-js) but that repo seems to have become stale.  

So again wondering if 

- do you still recommend that repo as deployment tool
- What other means is there if I want to make use NextJS. SSR etc, 

Thanks in advanced

**Comment by VeVarunSharma** [comment]
> Hi, I am currently exploring adopting Nextjs with terraform. Currently have a terraform based project that is using React.JS static assets to S3. I would like to switch over to Next.js to start benefiting of some of the features like SSR and wanted to know if there is any support fr it or example that can help me on that journey. I know on your site one of the deployment mechanisms mentions this repo [terraform-aws-next](https://github.com/milliHQ/terraform-aws-next-js) but that repo seems to have become stale.
> 
> So again wondering if
> 
> * do you still recommend that repo as deployment tool
> * What other means is there if I want to make use NextJS. SSR etc,
> 
> Thanks in advanced

Just saw this now mate!

Not sure if you found a solution already but if your project is just a static site, I don't see a a huge reason to switch over to next.js. There are some fantastic feature there but it be a larger effort to migrate & refactor your project to next for little ROI.
You can definitely write some custom terraform code where you can set up some infrastructure around S3 for AWS if that's where you want to deploy your static assets. 
Although I find using the Vercel platform a lot more dev friendly.

I hope this helps!
